File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/credit/port/UserCreditRepository.java
package com.study.webflux.rag.domain.credit.port;

import com.study.webflux.rag.domain.credit.model.UserCredit;
import com.study.webflux.rag.domain.dialogue.model.UserId;
import reactor.core.publisher.Mono;

public interface UserCreditRepository {

	Mono<UserCredit> findByUserId(UserId userId);

	Mono<UserCredit> save(UserCredit userCredit);
}

File: backend/domain/src/main/kotlin/com/miyou/app/domain/credit/port/UserCreditRepository.kt
package com.miyou.app.domain.credit.port

import com.miyou.app.domain.credit.model.UserCredit
import reactor.core.publisher.Mono

interface UserCreditRepository {
    fun findByUserId(userId: String): Mono<UserCredit>

    fun save(userCredit: UserCredit): Mono<UserCredit>
}
